pennylane/interfaces/execution.py

```diff
@@ -360,6 +360,22 @@ def cost_fn(params, x):
         for i, tape in enumerate(tapes):
             tapes[i] = expand_fn(tape)
 
+        if gradient_kwargs.get("method", "") == "adjoint_jacobian":
+            # adjoint jacobian needs further expansion
+            # this is a quick patch that should be cleaned up by later refactoring
+            if INTERFACE_MAP[interface] == "jax":
+                stop_at = (
+                    ~qml.operation.is_measurement & qml.operation.has_nopar | qml.operation.has_gen
+                )
+            else:
+                stop_at = (
+                    ~qml.operation.is_measurement & ~qml.operation.is_trainable
+                    | qml.operation.has_gen
+                )
+            for i, tape in enumerate(tapes):
+                if any(not stop_at(op) for op in tape.operations):
+                    tapes[i] = tape.expand(stop_at=stop_at, depth=10)

```

Comment:
Should the 10 be set as a function argument default? I worry magic values like this could cause issues somewhere.